### DISCUSSION TITLE
How to test Ebiten?

### DISCUSSION THREAD

**Discussion by xescugc** [hajimehoshi/ebiten#2835] [Q&A]
I was looking through the codebase/issues/discussions to see a way in which I could test Ebiten automatically. Basically I had in mind something that I could just say `MouseMovedTo(x,y)` or `ClickOn(x,y)` so then I could test my logic when those actions happen.

Is there a way currently to do this? And/Or which way would you recommend to test this?

**Comment by hajimehoshi** [comment]
Ebitengine itself doesn't provide a way to emulate inputting. So, you would have to create an abstract inputting later, make your game depend on it instead of Ebitengine's input APIs directly, and create an implementation to emulate inputs.
